collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -93,32 +100,21 @@
      */
     private boolean viewportInitialized;
 
-    @VisibleForTesting public ViewModelProvider.Factory viewModelFactory;
+    @VisibleForTesting
+    public ViewModelProvider.Factory viewModelFactory;
 
-    @Override public void onCreate(Bundle savedInstanceState) {
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         DaggerUtils.getComponent(this).inject(this);
 
-        FormsDao dao = new FormsDao();
-        Form form = null;
-        List<Form> forms = dao.getFormsFromCursor(dao.getFormsCursor(getIntent().getData()));
-        if (forms.size() == 1) {
-            form = forms.get(0);
-        } else if (viewModelFactory == null) {
```

Comment:
Don't we need this check anymore?

Reply:
I think for the form to not exist we'd have to have passed an incorrect ID in. Given that this Activity is only called from our app that'd be bad code rather than bad input, so I don't think we need to check for it. What do you think? 

I'm not sure I understand why the `else` is an `else if` with a check on `viewModelFactory` here.

Reply:
> I think for the form to not exist we'd have to have passed an incorrect ID in. Given that this Activity is only called from our app that'd be bad code rather than bad input, so I don't think we need to check for it. What do you think?

generally I agree ok let's try without that check. 

> I'm not sure I understand why the else is an else if with a check on viewModelFactory here.

It was added just to fix tests https://github.com/getodk/collect/pull/4337/commits/b396e9fc2fd378cc05fee71ddcf307fc1f2c3090